net_util/src/tap.rs

```diff
@@ -130,6 +130,7 @@ fn ipv6_mask_to_prefix(mask: Ipv6Addr) -> Result<u8> {
 
 impl Tap {
     unsafe fn ioctl_with_mut_ref<F: AsRawFd, T>(fd: &F, req: c_ulong, arg: &mut T) -> Result<()> {
+        // SAFETY: We trust the kernel and the parameters.
```

Comment:
It appears the main change from edition 2024 is that unsafe block is also required from unsafe functions, which makes sense.

A clearer way to document is adding two SAFETY comments:

* One for the unsafe function, explaining the contract that callers must uphold.
* Another for the unsafe block within it, justifying why the code is sound based on that contract, which should be trivial for our cases here.

Reply:
> A clearer way to document is adding two SAFETY comments:
>
> - One for the unsafe function, explaining the contract that callers must uphold.
> - Another for the unsafe block within it, justifying why the code is sound based on that > contract, which should be trivial for our cases here.


I agree! But to me this feels like a orthogonal issue which should be done in a dedicated PR. There is even a lint to require `# Safety` in the doc comments of a function. Is this okay for you?

Reply:
We need to have proper SAFETY comments for this PR, because this is what edition 2024 requires and what is missing from our code base. Also, [repeating mysel](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/7256#pullrequestreview-3145248533)f in case it is lost, let's add unsafe comments to the same commit when unsafe blocks are added.

Of course, if you feel more comfortable, feel free to use separate PRs to address these issues and make this PR smaller.

Reply:
Opened #7295 to resolve this discussion.